pkg/cephfs/mountcache.go

```diff
@@ -0,0 +1,333 @@
+package cephfs
+
+import (
+	"encoding/base64"
+	"os"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/ceph/ceph-csi/pkg/util"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+type volumeMountEntry struct {
+	NodeID        string `json:"nodeID"`
+	DriverName    string `json:"driverName"`
+	DriverVersion string `json:"driverVersion"`
+
+	Namespace string `json:"namespace"`
+
+	VolumeID      string            `json:"volumeID"`
+	Secrets       map[string]string `json:"secrets"`
+	StagingPath   string            `json:"stagingPath"`
+	TargetPaths   map[string]bool   `json:"targetPaths"`
+	CreateTime    time.Time         `json:"createTime"`
+	LastMountTime time.Time         `json:"lastMountTime"`
+	LoadCount     uint64            `json:"loadCount"`
+}
+
+type volumeMountCacheMap struct {
+	DriverName     string
+	DriverVersion  string
+	NodeID         string
+	MountFailNum   int64
+	MountSuccNum   int64
+	Volumes        map[string]volumeMountEntry
+	NodeCacheStore util.NodeCache
+	MetadataStore  util.CachePersister
+}
+
+var (
+	MountCacheDir          = ""
+	volumeMountCachePrefix = "cephfs-mount-cache-"
+	volumeMountCache       volumeMountCacheMap
+	volumeMountCacheMtx    sync.Mutex
+)
+
+func remountHisMountedPath(name string, v string, nodeID string, cachePersister util.CachePersister) error {
```

Comment:
`remountHisMountedPath` is a rather confusing name...how about `remountCachedVolumes`?

Reply:
Driver version is already defined as a constant in `driver.go`, no need to pass it in args here. Is `nodeID` even necessary?

Reply:
Also, imho this function does more than its name suggests. How about splitting it into `initVolumeMountCache` and `remountCachedVolumes`?

Reply:
> `remountHisMountedPath` is a rather confusing name...how about `remountCachedVolumes`?

`remountCachedVolumes` is better .

> Driver version is already defined as a constant in `driver.go`, no need to pass it via args here. Is `nodeID` even necessary?

i think if csi plugin  upgrade future,  some logic may need change  by tell different csi plugin version or driver name.

> Also, imho this function does more than its name suggests. How about splitting it into `initVolumeMountCache` and `remountCachedVolumes`?
ok.